components/todoist/actions/search-tasks/search-tasks.mjs

```diff
@@ -0,0 +1,64 @@
+import todoist from "../../todoist.app.mjs";
+
+export default {
+  key: "todoist-search-tasks",
+  name: "Search Tasks",
+  description: "Search tasks by name, label, project and/or section. [See Docs](https://developer.todoist.com/rest/v2/#get-active-tasks)",
+  version: "0.0.1",
+  type: "action",
+  props: {
+    todoist,
+    name: {
+      type: "string",
+      label: "Name Match",
+      description: "Returns tasks that contain the specified string in their name.",
+      optional: true,
+      default: "",
+    },
+    label: {
+      propDefinition: [
+        todoist,
+        "label",
+      ],
+    },
+    project: {
+      propDefinition: [
+        todoist,
+        "project",
+      ],
+    },
+    section: {
+      propDefinition: [
+        todoist,
+        "section",
+        (c) => ({
+          project: c.project,
+        }),
+      ],
+    },
+  },
+  async run ({ $ }) {
+    const {
+      name,
+      label,
+      project,
+      section,
+    } = this;
+    const tasks = await this.todoist.getActiveTasks({
+      $,
+      params: {
+        label: label,
+        project_id: project,
+        section_id: section,
+      },
+    });
+    let result = tasks.filter((task) => (task.content).includes(name));
+    let summary = `${result.length} task${result.length == 1
+      ? ""
+      : "s"} found`;
+    $.export("$summary", summary);
+    return result.length > 0
+      ? result
+      : undefined;
```

Comment:
For consistency, I would return the value itself even if it has zero length - so as to avoid breaking future steps that may rely on an array result from this component.
```suggestion
    return result;
```